Mathlib/Data/Fin/SuccPred.lean

```diff
@@ -43,6 +43,18 @@ theorem succ_eq {n : ℕ} : SuccOrder.succ = fun a => if a < Fin.last n then a +
 theorem succ_apply {n : ℕ} (a) : SuccOrder.succ a = if a < Fin.last n then a + 1 else a :=
```

Comment:
These should be about `Order.succ` too, and you should use them below. I think `SuccOrder.succ` is an implementation detail.

Reply:
I have redefined the `SuccOrder` and `PredOrder` using the `Fin.succ/pred/etc` API instead of arithmetic, and I have deprecated the lemmas `succ_eq/apply` (which are unused).